drivers/pwm/CMakeLists.txt

```diff
@@ -22,8 +22,9 @@ zephyr_library_sources_ifdef(CONFIG_PWM_SAM0_TCC	pwm_sam0_tcc.c)
 zephyr_library_sources_ifdef(CONFIG_PWM_NPCX		pwm_npcx.c)
 zephyr_library_sources_ifdef(CONFIG_PWM_XLNX_AXI_TIMER	pwm_xlnx_axi_timer.c)
 zephyr_library_sources_ifdef(CONFIG_PWM_MCUX_PWT 	pwm_mcux_pwt.c)
-zephyr_library_sources_ifdef(CONFIG_PWM_GECKO   pwm_gecko.c)
-zephyr_library_sources_ifdef(CONFIG_PWM_GD32 pwm_gd32.c)
+zephyr_library_sources_ifdef(CONFIG_PWM_GECKO		pwm_gecko.c)
+zephyr_library_sources_ifdef(CONFIG_PWM_GD32		pwm_gd32.c)
+zephyr_library_sources_ifdef(CONFIG_PWM_TEST		pwm_test.c)
```

Comment:
should we create a dedicated source tree somewhere here: https://github.com/zephyrproject-rtos/zephyr/tree/main/tests for such test code that are shared between tests ?

as to not pollute the general source code tree ?

Reply:
I think that's a good idea.